sdk/storage/azfile/service/client_test.go

```diff
@@ -253,29 +251,78 @@ func (s *ServiceUnrecordedTestsSuite) TestSASServiceClient() {
 	}
 	expiry := time.Now().Add(time.Hour)
 	sasUrl, err := serviceClient.GetSASURL(resources, permissions, expiry, nil)
-	_require.Nil(err)
+	_require.NoError(err)
 
 	svcClient, err := testcommon.GetServiceClientNoCredential(s.T(), sasUrl, nil)
-	_require.Nil(err)
-
-	// create share using SAS
-	//_, err = svcClient.CreateShare(context.Background(), shareName, nil)
-	//_require.Nil(err)
-	//
-	//_, err = svcClient.DeleteShare(context.Background(), shareName, nil)
-	//_require.Nil(err)
+	_require.NoError(err)
 
 	resp, err := svcClient.GetProperties(context.Background(), nil)
 	_require.NoError(err)
 	_require.NotNil(resp.RequestID)
+
+	// create share using account SAS
+	_, err = svcClient.CreateShare(context.Background(), shareName, nil)
+	_require.NoError(err)
+
+	defer func() {
+		_, err := svcClient.DeleteShare(context.Background(), shareName, nil)
```

Comment:
won't this defer Delete Share come back with an error since you are Deleting below?

Reply:
oh  nvm I see the restore share